src/main/kotlin/app/revanced/patches/youtube/misc/fix/playback/SpoofSignaturePatch.kt

```diff
@@ -45,22 +50,22 @@ object SpoofSignaturePatch : BytecodePatch(
                 .nextMethod(it.scanResult.patternScanResult!!.startIndex, true).getMethod() as MutableMethod
 
             setParamMethod.apply {
-                val protobufParameterRegister = 3
+                val videoIdParameter = 1
+                val protobufParameter = 3
 
                 addInstructions(
                     0,
                     """
-                        invoke-static {p$protobufParameterRegister}, $INTEGRATIONS_CLASS_DESCRIPTOR->spoofParameter(Ljava/lang/String;)Ljava/lang/String;
-                        move-result-object p$protobufParameterRegister
+                        invoke-static {p$videoIdParameter, p$protobufParameter}, $INTEGRATIONS_CLASS_DESCRIPTOR->spoofParameter(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
```

Comment:
Instead of this, make `VideoIdPatch` hook `videoIdParameter` earlier. You can hook the same method in addition to the current hook of `VideoIdPatch` so that `VideoInformationPatch` can be used in this patch. I believe the earliest useful time for a video id hook is from the player model response. One of the getters that returns the video id can be hooked which would act as the earliest point in time where the video id would be useful.

Reply:
Sorry, I am a little confused on the new commits. The only change that I had in mind was that the integrations field "videoId" is also set from the method that was previously used to retrieve the video id and pass to `spoofParameter`.

This way, the field would be set earlier in time, so that the patches such as the client spoof patch could use it.

Reply:
> The problem is the video id needs to be set before the protobuffer hook, and both of hooks are in the same method.

In this case a third patch can be used which both patches depend on. The third patch would manage this specific method. It could be called a generic name such as "PlayerResponseMethodHookPatch". The patch would offer two methods. One method to hook the video id string, and one method to hook the player parameter string. Then, both the video id patch as well as the spoof patch could make use of it. The video id patch would simply delegate its inject function to that of PlayerResponseMethodHookPatch and  the spoof patch would call the PlayerResponseMethodHookPatch to hook the player parameter. This way the single-responsibility OOP principle is retained.